x-pack/plugins/observability_solution/infra/server/routes/metrics_sources/index.ts

```diff
@@ -231,15 +237,18 @@ export const initMetricsSourceConfigurationRoutes = (libs: InfraBackendLibs) =>
             size: 0,
             ...(modules.length > 0
               ? {
-                  query: {
-                    bool: {
-                      should: [
-                        ...termsQuery(EVENT_MODULE, ...modules),
-                        ...termsQuery(METRICSET_MODULE, ...modules),
-                      ],
-                      minimum_should_match: 1,
+                  query: getDataTierFilterCombined({
+                    filter: {
+                      bool: {
+                        should: [
+                          ...termsQuery(EVENT_MODULE, ...modules),
+                          ...termsQuery(METRICSET_MODULE, ...modules),
+                        ],
+                        minimum_should_match: 1,
+                      },
                     },
-                  },
+                    excludedDataTiers,
+                  }),
```

Comment:
How about changing the `search` function to add this filter? This way, usages of `infraMetricsClient` won't have to worry about filtering out frozen tier, eg:

https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm_data_access/server/lib/helpers/create_es_client/create_apm_event_client/index.ts#L170

https://github.com/crespocarlos/kibana/blob/main/x-pack/plugins/observability_solution/apm/server/lib/helpers/get_apm_event_client.ts#L48